### PR TITLE
fix: ValueError when using FromOriginalModelMixin in subclasses #8440

### PR DESCRIPTION
**Pull request by fkcptlst**
(cherry picked from commit 92859978436acf844760fc0e992165b489d0180a)

# What does this PR do?

Fixes # (issue)

#8440

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.  #8440
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?  **test snippet below**

```py
import torch

from diffusers import AutoencoderKL


class MyVAE(AutoencoderKL):
    pass


if __name__ == "__main__":
    MyVAE.from_single_file("./checkpoints/realisticVisionV51_v51VAE.safetensors",
                           subfolder="vae",
                           torch_dtype=torch.float16,
                           use_safetensors=True)

    print("done.")
```

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@DN6

### CODE REVIEW

File: src/diffusers/loaders/single_file_model.py [hunk 1]
```diff
@@ -67,6 +67,23 @@
...
+    """
+    Find if src_cls is a subclass of dst_cls whose name is dst_cls_str
+    """
+    for cls in src_cls.__mro__:
+        if cls.__name__ == dst_cls_str:
+            return True
+    return False
+
+
+def _get_single_file_loadable_mapping_class(cls):
```
Comment (DN6) [inline src/diffusers/loaders/single_file_model.py:80]: Nice! I think we can leverage Python's built in `issubclass` here if we use 

```suggestion
def _get_single_file_loadable_mapping_class(cls):
    diffusers_module = importlib.import_module(__name__.split(".")[0])
    for loadable_class_str in SINGLE_FILE_LOADABLE_CLASSES:
        loadable_class = getattr(diffusers_module, loadable_class_str)

        if issubclass(cls, loadable_class):
            return loadable_class_str

    return None
```

Then we just need a single function to do the check.

Comment (fkcptlst) [reply src/diffusers/loaders/single_file_model.py:80]: Sure! This is a more elegant approach indeed.

**Comment by fkcptlst** [comment]
I made 2 commits to 1. resolve the review, 2. remove the redundant code fragments of the original code

File: src/diffusers/loaders/single_file_model.py [hunk 2]
```diff
@@ -67,6 +67,17 @@
 }
 
 
+def _get_single_file_loadable_mapping_class(cls):
+    diffusers_module = importlib.import_module(__name__.split(".")[0])
```
Comment (DN6) [inline src/diffusers/loaders/single_file_model.py:71]: Thanks. I think we just need to add `import importlib` into the file 👍🏽

Comment (fkcptlst) [reply src/diffusers/loaders/single_file_model.py:72]: you are right

**Comment by DN6** [comment]
HI @fkcptlst will resolve the issues with the CI and merge this soon.

**Review by DN6 — APPROVED** [review]

**Pull request merged**